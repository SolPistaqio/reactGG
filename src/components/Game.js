import { React, Component, createContext } from "react";
import Router from "./UI/router";
import { createGhosts } from "./util/ghostFactory";

export const GameContext = createContext({});

class Game extends Component {
  constructor(props) {
    super(props);
    this.state = {
      round: 1,
      coins: 10,
      userTeam: [],
      currentView: "start",
      activeGhost: {},
      toggleView: this.toggleView,
      teamAdd: this.teamAdd,
      ghostBuy: this.ghostBuy,
      pickActiveGhost: this.pickActiveGhost,
      abortMarketOperation: this.abortMarketOperation,
    };
  }

  abortMarketOperation = () => {
    this.setState({ activeGhost: {} });
    this.toggleView("game");
  };
  toggleView = (newView) => {
    console.log(`Navigating to ${newView}`);
    this.setState({ currentView: newView });
  };
  coinOperation(amountToAdd) {
    this.setState({ coins: this.state.coins + amountToAdd });
  }
  teamAdd = (ghost) => {
    console.log(ghost);
    this.setState({ userTeam: [...this.state.userTeam, ...ghost] });
  };
  teamReplace(replacee, replacement) {
    const team = this.state.userTeam;
    const order = team.indexOf(replacee);
    team[order] = replacement;
    this.setState({ userTeam: team });
  }
  pickActiveGhost = (ghost) => {
    console.log(ghost);
    this.setState({ activeGhost: ghost });
  };

  ghostBuy = (ghost) => {
    if (this.state.activeGhost.type) {
      this.teamReplace(this.state.activeGhost, ghost);
      this.setState({ activeGhost: {} });
    } else {
      this.teamAdd([ghost]);
    }
    this.coinOperation(0 - ghost.price);
    this.toggleView("game");
  };

  componentDidMount() {
    if (this.state.round === 1) {
      this.teamAdd(createGhosts(2));
    }
  }

  render() {
    return (
      <GameContext.Provider value={this.state}>
        <Router view={this.state.currentView} />
      </GameContext.Provider>
    );
  }
}

export default Game;
