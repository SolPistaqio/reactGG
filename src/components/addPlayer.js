import React, { Component } from "react";
import { Icon } from "@iconify/react";
import "./css/addButton.css";

class AddGhost extends Component {
  openMarket(e) {
    console.log("open it!");
  }
  render() {
    return (
      <Icon
        icon="carbon:add-filled"
        height="100px"
        className="add"
        onClick={() => this.openMarket()}
      />
    );
  }
}

export default AddGhost;
