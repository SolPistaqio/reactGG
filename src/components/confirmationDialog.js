import React, { Component } from "react";
import "./css/confirmation.css";

class Confirmation extends Component {
  render() {
    return (
      <div className="modal">
        <div className="modal-content">
          <h1>{this.props.text}</h1>

          <button>Confirm</button>
          <button>Cancel</button>
        </div>
      </div>
    );
  }
}

export default Confirmation;
