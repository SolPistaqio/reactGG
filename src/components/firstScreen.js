import { Component } from "react";

class Start extends Component {
  render() {
    return (
      <div style={{ display: "grid", placeItems: "center" }}>
        <img src="/GG30.gif" alt="GhostGame logo" />
        <button>START</button>
      </div>
    );
  }
}

export default Start;
