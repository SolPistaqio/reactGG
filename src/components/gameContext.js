import { createContext } from "react";

const GameContext = createContext({
  coins: 10,
  userTeam: [],
  currentView: "start",
  activeGhost: {},
  toggleView: () => {},
  coinOperation: () => {},
  teamAdd: () => {},
  teamRepalce: () => {},
  pickActiveGhost: () => {},
});

// export default GameContext;
