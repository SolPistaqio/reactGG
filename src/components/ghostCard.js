import React, { Component } from "react";
import Ghost3 from "./characters/Ghost3";
import { Icon } from "@iconify/react";

class GhostCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      type: 1,
      maxHealth: 3,
      color1: "",
      color2: "",
      currentHealth: 2,
    };
  }

  getGhostColor = () => {
    const letters = "0123456789ABCDEF";
    let ghostColor = "#";
    for (var i = 0; i < 6; i++) {
      ghostColor += letters[Math.floor(Math.random() * 16)];
    }
    return ghostColor;
  };

  setGhostColors = () => {
    const col1 = this.getGhostColor();
    const col2 = this.getGhostColor();
    this.setState({ color1: col1, color2: col2 });
  };

  renderMissingHealth = () => {
    if (this.state.currentHealth < this.state.maxHealth) {
      const numOfHearts = this.state.maxHealth - this.state.currentHealth;
      for (let i = 0; i < numOfHearts; i++) {
        return <Icon icon="ant-design:heart-outlined" height="30px" />;
      }
    }
  };

  componentWillMount() {
    this.setGhostColors();
  }

  render() {
    const healthBar = [];
    for (let i = 0; i < this.state.currentHealth; i++) {
      healthBar.push("heart-filled");
    }
    if (this.state.currentHealth < this.state.maxHealth) {
      const numOfHearts = this.state.maxHealth - this.state.currentHealth;
      for (let i = 0; i < numOfHearts; i++) {
        healthBar.push("heart-outlined");
      }
    }
    return (
      <div
        style={{
          display: "inline-grid",
          width: "250px",
          placeItems: "center",
        }}
      >
        <Ghost3
          color1={this.state.color1}
          color2={this.state.color2}
          style={{
            display: "block",
          }}
        />
        <div
          style={{
            display: "block",
          }}
        >
          {healthBar.map((heart, index) => (
            <Icon
              icon={"ant-design:" + heart}
              height="50px"
              color="red"
              key={index}
            />
          ))}
        </div>
      </div>
    );
  }
}

export default GhostCard;
