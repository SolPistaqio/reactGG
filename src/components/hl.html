 
                                                    <a href="/citizenship-austria/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/267/047/ff8/thumb__28_0_0_0_auto.png" alt="Австрия">Австрия</a>
                                                    <a href="/citizenship-antigua/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/d51/847/f84/thumb__28_0_0_0_auto.png" alt="Антигуа и Барбуда">Антигуа и Барбуда</a>
                                                    <a href="/citizenship-grenada/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/4fe/43f/343/thumb__28_0_0_0_auto.png" alt="Гренада">Гренада</a>
                                                    <a href="/citizenship-dominica/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/580/a66/237/thumb__28_0_0_0_auto.png" alt="Доминика">Доминика</a>
                                                    <a href="/citizenship-jordan/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/8fe/063/16f/thumb__28_0_0_0_auto.png" alt="Иордания">Иордания</a>
                                                    <a href="/citizenship-malta/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/417/22d/4b2/thumb__28_0_0_0_auto.png" alt="Мальта">Мальта</a>
                                                    <a href="/citizenship-north-macedonia/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/8e8/542/449/thumb__28_0_0_0_auto.png" alt="Северная Македония">Северная Македония</a>
                                                    <a href="/citizenship-skn/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/f51/ae3/c73/thumb__28_0_0_0_auto.png" alt="Сент-Китс и Невис">Сент-Китс и Невис</a>
                                                    <a href="/citizenship-stlucia/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/cb3/671/7a1/thumb__28_0_0_0_auto.png" alt="Сент-Люсия">Сент-Люсия</a>
                                                    <a href="/citizenship-turkey/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/f9c/75f/9a6/thumb__28_0_0_0_auto.png" alt="Турция">Турция</a>
                                                    <a href="/citizenship-montenegro/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/b74/e0c/000/thumb__28_0_0_0_auto.png" alt="Черногория">Черногория</a>
                                                            <a href="/residence-australia/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/155/2ba/011/thumb__28_0_0_0_auto.png" alt="Австралия">Австралия</a>
                                                            <a href="/residence-austria/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/267/047/ff8/thumb__28_0_0_0_auto.png" alt="Австрия">Австрия</a>
                                                            <a href="/residence-uk/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/12b/839/d48/thumb__28_0_0_0_auto.png" alt="Великобритания">Великобритания</a>
                                                            <a href="/residence-hong-kong/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/48b/1c4/592/thumb__28_0_0_0_auto.png" alt="Гонконг">Гонконг</a>
                                                            <a href="/residence-greece/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/e9b/8a5/dc4/thumb__28_0_0_0_auto.png" alt="Греция">Греция</a>
                                                            <a href="/residence-jersey/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/822/f98/64e/thumb__28_0_0_0_auto.png" alt="Джерси">Джерси</a>
                                                            <a href="/residence-spain/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/fa1/f96/27f/thumb__28_0_0_0_auto.png" alt="Испания">Испания</a>
                                                            <a href="/residence-italy/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/60d/964/0f4/thumb__28_0_0_0_auto.png" alt="Италия">Италия</a>
                                                            <a href="/residence-canada/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/42d/d95/8fb/thumb__28_0_0_0_auto.png" alt="Канада">Канада</a>
                                                            <a href="/residence-cyprus/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/a1c/d26/b3a/thumb__28_0_0_0_auto.png" alt="Кипр">Кипр</a>
                                                            <a href="/residence-latvia/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/db5/4ac/cff/thumb__28_0_0_0_auto.png" alt="Латвия">Латвия</a>
                                                            <a href="/residence-luxembourg/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/bea/f0a/2e2/thumb__28_0_0_0_auto.png" alt="Люксембург">Люксембург</a>
                                                            <a href="/residence-mauritius/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/6e8/756/d66/thumb__28_0_0_0_auto.png" alt="Маврикий">Маврикий</a>
                                                            <a href="/residence-malaysia/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/0ac/862/c1e/thumb__28_0_0_0_auto.png" alt="Малайзия">Малайзия</a>
                                                            <a href="/residence-malta/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/417/22d/4b2/thumb__28_0_0_0_auto.png" alt="Мальта">Мальта</a>
                                                            <a href="/residence-monaco/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/0e6/570/117/thumb__28_0_0_0_auto.png" alt="Монако">Монако</a>
                                                            <a href="/residence-new-zealand/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/917/155/f0c/thumb__28_0_0_0_auto.png" alt="Новая Зеландия">Новая Зеландия</a>
                                                            <a href="/residence-united-arab-emirates/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/168/9e7/f5d/thumb__28_0_0_0_auto.png" alt="ОАЭ">ОАЭ</a>
                                                            <a href="/residence-panama/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/616/69b/b29/thumb__28_0_0_0_auto.png" alt="Панама">Панама</a>
                                                            <a href="/residence-portugal-golden-visa/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/d9d/602/34d/thumb__28_0_0_0_auto.png" alt="Португалия">Португалия</a>
                                                            <a href="/residence-singapore/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/88d/76b/1cd/thumb__28_0_0_0_auto.png" alt="Сингапур">Сингапур</a>
                                                            <a href="/residence-usa/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/10d/e62/6c3/thumb__28_0_0_0_auto.png" alt="США">США</a>
                                                            <a href="/residence-thailand/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/3c7/82d/97c/thumb__28_0_0_0_auto.png" alt="Таиланд">Таиланд</a>
                                                            <a href="/permanent-residence-switzerland/"><img loading="lazy" class="card-crest" src="https://ru.henleyglobal.com/storage/app/uploads/public/b15/69c/cd3/thumb__28_0_0_0_auto.png" alt="Швейцария">Швейцария</a>
                                                        <a href="/citizenship-austria/">Австрия</a>
                                                        <a href="/citizenship-antigua/">Антигуа и Барбуда</a>
                                                        <a href="/citizenship-grenada/">Гренада</a>
                                                        <a href="/citizenship-dominica/">Доминика</a>
                                                        <a href="/citizenship-jordan/">Иордания</a>
                                                        <a href="/citizenship-malta/">Мальта</a>
                                                        <a href="/citizenship-north-macedonia/">Северная Македония</a>
                                                        <a href="/citizenship-skn/">Сент-Китс и Невис</a>
                                                        <a href="/citizenship-stlucia/">Сент-Люсия</a>
                                                        <a href="/citizenship-turkey/">Турция</a>
                                                        <a href="/citizenship-montenegro/">Черногория</a>
                                                        <a href="/residence-australia/">Австралия</a>
                                                        <a href="/residence-austria/">Австрия</a>
                                                        <a href="/residence-uk/">Великобритания</a>
                                                        <a href="/residence-hong-kong/">Гонконг</a>
                                                        <a href="/residence-greece/">Греция</a>
                                                        <a href="/residence-jersey/">Джерси</a>
                                                        <a href="/residence-spain/">Испания</a>
                                                        <a href="/residence-italy/">Италия</a>
                                                        <a href="/residence-canada/">Канада</a>
                                                        <a href="/residence-cyprus/">Кипр</a>
                                                        <a href="/residence-latvia/">Латвия</a>
                                                        <a href="/residence-luxembourg/">Люксембург</a>
                                                        <a href="/residence-mauritius/">Маврикий</a>
                                                        <a href="/residence-malaysia/">Малайзия</a>
                                                        <a href="/residence-malta/">Мальта</a>
                                                        <a href="/residence-monaco/">Монако</a>
                                                        <a href="/residence-new-zealand/">Новая Зеландия</a>
                                                        <a href="/residence-united-arab-emirates/">ОАЭ</a>
                                                        <a href="/residence-panama/">Панама</a>
                                                        <a href="/residence-portugal-golden-visa/">Португалия</a>
                                                        <a href="/residence-singapore/">Сингапур</a>
                                                        <a href="/residence-usa/">США</a>
                                                        <a href="/residence-thailand/">Таиланд</a>
                                                        <a href="/permanent-residence-switzerland/">Швейцария</a>
                       
          