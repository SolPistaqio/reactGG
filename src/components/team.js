import React, { Component } from "react";

import GhostCard from "./ghostCard";
import AddGhost from "./addPlayer";

import "./css/team.css";

class Team extends Component {
  state = {
    team: [
      {
        type: 1,
        maxHealth: 3,
        color1: "",
        color2: "",
        currentHealth: 2,
      },
      {
        type: 2,
        maxHealth: 2,
        color1: "",
        color2: "",
        currentHealth: 2,
      },
      {
        type: 3,
        maxHealth: 2,
        color1: "",
        color2: "",
        currentHealth: 1,
      },
    ],
  };
  render() {
    const playerTeam = this.state.team;
    const needsToAdd = 3 - playerTeam.length;
    let addButtons = "";
    if (needsToAdd > 0) {
      addButtons = [...Array(needsToAdd)].map((e, index) => (
        <AddGhost key={index} />
      ));
    }
    return (
      <div className="team">
        <h1
          style={{ color: "white", textAlign: "center", gridColumn: "1 / 4" }}
        >
          YOUR TEAM
        </h1>

        {playerTeam.map((ghost, index) => {
          return <GhostCard ghost={ghost} key={index} />;
        })}

        {addButtons}
      </div>
    );
  }
}

export default Team;
