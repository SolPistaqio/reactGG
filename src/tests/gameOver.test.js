import renderer from "react-test-renderer";
import GameOver from "../components/views/gameOver";
import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { GameContext } from "../components/context";

it("renders correctly", () => {
  const startNewGame = jest.fn();
  const toggleView = jest.fn();
  const tree = renderer
    .create(
      <GameContext.Provider value={{ toggleView, startNewGame, gameWon: true }}>
        <GameOver />
      </GameContext.Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly", () => {
  const startNewGame = jest.fn();
  const toggleView = jest.fn();
  const tree = renderer
    .create(
      <GameContext.Provider
        value={{ toggleView, startNewGame, gameWon: false }}
      >
        <GameOver />
      </GameContext.Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe("GameOver component", () => {
  it("should render the correct image and button based on the gameWon prop", () => {
    const startNewGame = jest.fn();
    const toggleView = jest.fn();

    const { getByAltText, getByText } = render(
      <GameContext.Provider value={{ toggleView, startNewGame, gameWon: true }}>
        <GameOver />
      </GameContext.Provider>
    );

    const image = getByAltText("Final screen");
    expect(image).toHaveAttribute("src", "/victory.gif");
    expect(getByText("START AGAIN")).toBeInTheDocument();
  });

  it("should render the correct image and button based on the gameWon prop", () => {
    const startNewGame = jest.fn();
    const toggleView = jest.fn();

    const { getByAltText, getByText } = render(
      <GameContext.Provider
        value={{ toggleView, startNewGame, gameWon: false }}
      >
        <GameOver />
      </GameContext.Provider>
    );

    const image = getByAltText("Final screen");
    expect(image).toHaveAttribute("src", "/defeat.gif");
    expect(getByText("START AGAIN")).toBeInTheDocument();
  });

  it("should call startNewGame and toggleView when the button is clicked", () => {
    const startNewGame = jest.fn();
    const toggleView = jest.fn();

    const { getByText } = render(
      <GameContext.Provider value={{ toggleView, startNewGame, gameWon: true }}>
        <GameOver />
      </GameContext.Provider>
    );

    const button = getByText("START AGAIN");
    fireEvent.click(button);

    expect(startNewGame).toHaveBeenCalled();
    expect(toggleView).toHaveBeenCalledWith("game");
  });
});
