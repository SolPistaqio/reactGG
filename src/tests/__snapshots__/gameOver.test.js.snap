// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="heroWrapper"
>
  <img
    alt="Final screen"
    className="hero"
    src="/victory.gif"
  />
  <button
    className="gameButton"
    onClick={[Function]}
  >
    START AGAIN
  </button>
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="heroWrapper"
>
  <img
    alt="Final screen"
    className="hero"
    src="/defeat.gif"
  />
  <button
    className="gameButton"
    onClick={[Function]}
  >
    START AGAIN
  </button>
</div>
`;
